### ISSUE TITLE
Set upper limit to attachment size

### ISSUE THREAD

**Issue by manuerwin** [spliit-app/spliit#84]
The app seems to allow any size file to be uploaded, strongly suggest something like a setting that optionally limits the max attachment size and displays an error/warning if exceeded to:
a) conserve storage and
b) prevent bad actors

See screenshot of the 35MB sample image I was able to upload just now (I've since deleted the expense, however I'm unsure if the image has been deleted from S3?)
![screencapture-spliit-app-groups-expenses-edit-2024-02-05-19_02_33](https://github.com/spliit-app/spliit/assets/982158/1c135c41-7121-47bd-bc8d-c75c7fa3875a)

**Comment by scastiel** [comment]
Agree! I found a nice way to add a limit for the file size when using next-s3-upload (the third package I used), but it requires a change in the library ([see this open PR](https://github.com/ryanto/next-s3-upload/pull/179)).

**Comment by manuerwin** [comment]
Seems like that will work, for S3 which is great. 
However if the size check happened in app/window, then it would be applicable to *any* file provider right?

**Comment by scastiel** [comment]
You’re right, I just pushed a check for the file size on the client side, which is nice for user experience anyway. But ultimately, we need to perform a better check, as for now nothing prevents a user from tweaking the code in the browser to disable the check.

**Comment by manuerwin** [comment]
After a bit more thought, suggest perhaps a max limit of 2MB?
That should allow for fairly large images and certainly PDF’s etc IMO.
The math on 5MB X many expenses => large over the long term IMO

**Comment by shynst** [comment]
> After a bit more thought, suggest perhaps a max limit of 2MB? That should allow for fairly large images and certainly PDF’s etc IMO. The math on 5MB X many expenses => large over the long term IMO

A setting you should probably be able to tweak in a .env file, instead of having to hardcode.

**Comment by manuerwin** [comment]
Great point about the .env file, agree!

**Comment by alfonzjanfrithz** [comment]
I tried to use the scan receipt straight from phone. my phone take it at 8mb. So it wouldnt work unless i trick it to snap it in whatsapp(that compress it first)

So, one idea is to do the compress before upload. This will massively reduce upload time, storage cost and potentially speed up any downstream image analysis.

Probably could do some resize in client side, convert to something like WebP. I know some image have metadata like location etc, idk how much, but it could contribute to file size i reckon. so if we can remove this kind of thing, that could probably reduce.

In an event client cant compress. Then only put the limit and do some validation above.
